src/Jackett.Common/Indexers/MoreThanTV.cs

```diff
@@ -246,33 +246,37 @@ private async Task GetReleases(ICollection<ReleaseInfo> releases, TorznabQuery q
 
         private ReleaseInfo GetReleaseInfo(IElement row, IElement downloadAnchor, string title, int category)
         {
-            // Parse required data
             var downloadAnchorHref = downloadAnchor.Attributes["href"].Value;
             var torrentId = downloadAnchorHref.Substring(downloadAnchorHref.LastIndexOf('=') + 1);
+            if (torrentId.Contains('#'))
+                torrentId = torrentId.Split('#')[0];
```

Comment:
I've noticed a few different places where we're extracting parameters out of the query portion of Urls recently for use in building a release. This kind of extraction becomes a lot more readable and resistant to changes in the website if we code it as follows:

```
var downloadAnchorHref = new Uri(downloadAnchor.Attributes["href"].Value);
var qParams = HttpUtility.ParseQueryString(downloadAnchorHref.Query);
var torrentId = qParams["paramName"];
```

`qParams` in the above code is a `NameValueCollection` like we currently use to build query strings in requests. This way, we call the query parameter by name, and if the website happens to adjust the Uri (either by adding or moving around parts of the Url or query) it won't break our parser.

Reply:
IMHO sites don't change often and when they do, the change the whole layout, not just one link. So, you have to rewrite part of the code anyway.